Source/SwiftLintFramework/Rules/Lint/IdenticalOperandsRule.swift

```diff
@@ -55,152 +56,54 @@ public struct IdenticalOperandsRule: ConfigurationProviderRule, OptInRule {
+        return Visitor(viewMode: .sourceAccurate)
+            .walk(tree: tree, handler: \.violationPositions)
+            .map { position in
+                StyleViolation(ruleDescription: Self.description,
+                               severity: configuration.severity,
+                               location: Location(file: file, position: position))
             }
-
-            return (Operand(index: leftTokenIndex - leftTokens.count + 1, tokens: leftTokens),
-                    Operand(index: rightTokenIndex, tokens: rightTokens))
     }
 }
 
-private extension StringView {
-    func subStringWithSyntaxToken(_ syntaxToken: SwiftLintSyntaxToken) -> String? {
-        return substringWithByteRange(syntaxToken.range)
-    }
-
-    func subStringBetweenTokens(_ startToken: SwiftLintSyntaxToken, _ endToken: SwiftLintSyntaxToken) -> String? {
-        let byteRange = ByteRange(location: startToken.range.upperBound,
-                                  length: endToken.offset - startToken.range.upperBound)
-        return substringWithByteRange(byteRange)
-    }
-
-    func isDotOrOptionalChainingBetweenTokens(_ startToken: SwiftLintSyntaxToken,
-                                              _ endToken: SwiftLintSyntaxToken) -> Bool {
-        return isRegexBetweenTokens(startToken, #"[\?!]?\."#, endToken)
+private extension SourceFileSyntax {
+    func folded() -> SourceFileSyntax? {
+        OperatorTable.standardOperators
+            .foldAll(self) { _ in }
+            .as(SourceFileSyntax.self)
     }
+}
 
-    func isWhiteSpaceBetweenTokens(_ startToken: SwiftLintSyntaxToken,
-                                   _ endToken: SwiftLintSyntaxToken) -> Bool {
-        guard let betweenTokens = subStringBetweenTokens(startToken, endToken) else { return false }
-        let range = betweenTokens.fullNSRange
-        return regex(#"^[\s\(,]*$"#).matches(in: betweenTokens, options: [], range: range).isNotEmpty
-    }
+private extension IdenticalOperandsRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
 
-    func isRegexBetweenTokens(_ startToken: SwiftLintSyntaxToken, _ regexString: String,
-                              _ endToken: SwiftLintSyntaxToken) -> Bool {
-        guard let betweenTokens = subStringBetweenTokens(startToken, endToken) else { return false }
+        override func visitPost(_ node: InfixOperatorExprSyntax) {
+            guard let operatorNode = node.operatorOperand.as(BinaryOperatorExprSyntax.self),
+                  IdenticalOperandsRule.operators.contains(operatorNode.operatorToken.withoutTrivia().text) else {
+                return
+            }
 
-        let range = betweenTokens.fullNSRange
-        return regex("^\\s*\(regexString)\\s*$").matches(in: betweenTokens, options: [], range: range).isNotEmpty
+            if node.leftOperand.withoutTrivia().description == node.rightOperand.withoutTrivia().description {
```

Comment:
I guess this should check that the operands are simple identifiers?

Reply:
I can only think of contrived examples, but if you did `random() == random()` it would be odd for the linter to tell you both sides are the same.

Maybe you’re right though that the value provided from catching legitimate violations outweighs the annoyance of contrived false positives.

Reply:
What about examples like `f(i: 3) == f(i:  3)` or `1+1 == 1 +  1`?

Reply:
@SimplyDanny I think the first one is currently covered by the rule. Not sure about the second one.

Reply:
The code highlight ignores multiple spaces in a row. That's why both sides in the first example looked the same. I've updated it to make the whitespace differences visible.

My point is that when only the string representation of both sides are compared, whitespace has an influence even though they are semantically equal.

Is it possible to compare the ASTs of both sides?